mlrun/runtimes/kubejob.py

```diff
@@ -56,19 +57,46 @@ def is_deployed(self):
             return True
         return False
 
-    def with_source_archive(self, source, pythonpath=None, pull_at_runtime=True):
+    def with_source_archive(
+        self, source, workdir=None, handler=None, pull_at_runtime=True
+    ):
         """load the code from git/tar/zip archive at runtime or build
 
         :param source:     valid path to git, zip, or tar file, e.g.
                            git://github.com/mlrun/something.git
                            http://some/url/file.zip
-        :param pythonpath: python search path relative to the archive root or absolute (e.g. './subdir')
+        :param handler: default function handler
+        :param workdir: working dir relative to the archive root or absolute (e.g. './subdir')
         :param pull_at_runtime: load the archive into the container at job runtime vs on build/deploy
         """
-        self.spec.build.load_source_on_run = pull_at_runtime
         self.spec.build.source = source
-        if pythonpath:
-            self.spec.pythonpath = pythonpath
+        if handler:
+            self.spec.default_handler = handler
+        if workdir:
+            self.spec.workdir = workdir
+        self.set_build_mode(pull_at_runtime)
+
+    def set_build_mode(self, pull_at_runtime):
+        """Set the source build/clone mode (in deploy or during runtime)
+
+        make sure the flags and image are aligned with the build mode
+
+        :param pull_at_runtime:  True - will clone the source code into the container at job runtime
+                                 False - will clone the code into the container image during build time
+        """
+        self.spec.build.load_source_on_run = pull_at_runtime
```

Comment:
So we have 3 names for the same exact thing - `pull_at_runtime`, `load_source_on_run` and `build_mode`
let's try to keep things simple, one thing, one name
Also if it's simply something that needs to happen on every set of an attribute, maybe just make this logic as the setter of this attribute

Reply:
ok for setter, should i replace `pull_at_runtime` with `load_source_on_run ` (break compatibility) ?